source/git/cli.go

```diff
@@ -0,0 +1,146 @@
+package git
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/moby/buildkit/session/networks"
+	"github.com/moby/buildkit/util/progress/logs"
+	"github.com/pkg/errors"
+)
+
+// gitCLI carries config to pass to the git CLI to make running multiple
```

Comment:
@vito any chance you could pull this git refactor into a separate PR? I was looking at unifying the buildx and buildkit git helpers into one place, and this would be a good start :tada: 

If you're too busy, I'm happy to pick it up :smile:

Reply:
Started hacking on this in https://github.com/moby/buildkit/pull/4106 :tada: (thanks!)